### ISSUE TITLE
Default value of keyword parameter in custom Protocol is not used

### ISSUE THREAD

**Issue by AndrewUshakov** [python/mypy#10484]
**Bug Report**

Consider the code below:

```Python
from typing import Protocol


class P(Protocol):
    def foo(self, n: int = 666) -> None: ...


class F:
    def foo(self, n: int = 666) -> None: ...


class S:
    def foo(self, n: int = 999) -> None: ...


def bar(x: P) -> None: ...


bar(F())
bar(S())
```

**Expected Behavior**

Warning in the last code line about incompatibility of class `S` and protocol `P`, because the default value of parameter `n` in `P.foo` is not ellipsis and the numerical value of `n` in `S.foo`  is different.

**Actual Behavior**

Mypy does not generate any warnings.

**Your Environment**

- Mypy version used:

8.12
0.820+dev.8642b351cff94c471333256f9cd6867807384f01

- Mypy command-line flags: 

none

- Mypy configuration options from `mypy.ini` (and other config files):

```
[mypy]

pretty = True
show_error_codes = True
strict = True
disallow_any_unimported = True
disallow_any_expr = True
disallow_any_decorated = True
disallow_any_explicit = True
no_warn_no_return = True
warn_unreachable = True
```

- Python version used:

3.8.10 (64-bit), 3.9.5 (64-bit), 3.10 beta  (64-bit)

- Operating system and version:

Windows 10 64-bit

**Comment by JelleZijlstra** [comment]
Can you clarify why you think that a different default value should make the Protocol incompatible? Default values aren't part of the interface, only the fact that there is a default value is.

**Comment by AndrewUshakov** [comment]
Just because I did not find any information about it in PEP 544 and mypy documentation... Ok, but doesn't that mean that any defaults are equivalent to `...`? If so, maybe any other default values in protocols should be prohibited, except cases mentioned [here](https://mypy.readthedocs.io/en/stable/stubs.html?highlight=default%20value#using-stub-file-syntax-at-runtime)?

**Comment by JelleZijlstra** [comment]
I guess it just hasn't come up. I'd welcome better documentation to make this explicit. Warning about non-`...` defaults in protocols is probably also a good idea, but might have some backward compatibility issues.

**Comment by AndrewUshakov** [comment]
>  Warning about non-... defaults in protocols is probably also a good idea, but might have some backward compatibility issues.

Yes, you are right. It can occur if your program need to be compatible both with Python 3.X and Python 2.X. In this case code will look like:

```Python
import sys

if sys.version_info >= (3, 8):
    from typing import TYPE_CHECKING, Protocol
    if TYPE_CHECKING:
        class P(Protocol):
            def foo(self, n=0):
                # type: (int) -> None
                pass
```
just because `...` causes a syntax error in Python 2.X :( I do not know if mypy can recognize such situation?
